src/storage/txn/scheduler.rs

```diff
@@ -364,17 +372,27 @@ impl<E: Engine> Scheduler<E> {
         let cb = Box::new(move |(cb_ctx, snapshot)| {
             executor.execute(cb_ctx, snapshot, task);
         });
-        if let Err(e) = self.engine.async_snapshot(&ctx, cb) {
-            SCHED_STAGE_COUNTER_VEC
-                .with_label_values(&[tag, "async_snapshot_err"])
-                .inc();
 
-            error!("engine async_snapshot failed"; "err" => ?e);
-            self.finish_with_err(cid, e.into());
+        let f = |engine: &E| {
+            if let Err(e) = engine.async_snapshot(&ctx, cb) {
+                SCHED_STAGE_COUNTER_VEC
+                    .with_label_values(&[tag, "async_snapshot_err"])
+                    .inc();
+
+                error!("engine async_snapshot failed"; "err" => ?e);
+                self.finish_with_err(cid, e.into());
+            } else {
+                SCHED_STAGE_COUNTER_VEC
+                    .with_label_values(&[tag, "snapshot"])
+                    .inc();
+            }
+        };
+
+        if let Some(engine) = self.engine.as_ref() {
```

Comment:
Don't understand why we need this logic.

Reply:
If it's called in scheduler worker threads, this field will be `None`.

Reply:
what is the probability that we run in scheduler worker threads?